pointcloud/basic_octree_utils_test.go

```diff
@@ -206,27 +208,42 @@ func validateBasicOctree(t *testing.T, bOct *BasicOctree, center r3.Vector, side
 				numLeafNodeEmptyNodes++
 			}
 
-			size += validateBasicOctree(t, child, r3.Vector{
+			childSize, childMaxProb := validateBasicOctree(t, child, r3.Vector{
 				X: center.X + i*sideLength/4.,
 				Y: center.Y + j*sideLength/4.,
 				Z: center.Z + k*sideLength/4.,
 			}, sideLength/2.)
+			size += childSize
+
+			if math.IsNaN(bOct.node.maxVal) {
+				maxVal = math.Max(bOct.node.maxVal, childMaxProb)
+			} else {
+				maxVal = math.Max(maxVal, childMaxProb)
+			}
 		}
 		test.That(t, size, test.ShouldEqual, bOct.size)
+		if math.IsNaN(bOct.node.maxVal) {
```

Comment:
[q] does `test.ShouldEqual` not work on `NaN`s? If they do this if statement shouldn't be necessary and if they dont we can still remove if statement and do ShouldResemble instead